Source/components/Page.js

```diff
@@ -0,0 +1,147 @@
+import { DefinedPropertiesKey } from '../WrappedObject'
+import { Group } from './Group'
+import { Selection } from '../Selection'
+import { Rectangle } from '../Rectangle'
+import { Types } from '../enums'
+import { Factory } from '../Factory'
+import { DEFAULT_EXPORT_OPTIONS } from './Layer'
+import { wrapNativeObject, wrapObject } from '../wrapNativeObject'
+
+/**
+ * Represents a Page in a Sketch document.
+ */
+export class Page extends Group {
+  /**
+   * Make a new page object.
+   *
+   * @param {MSPage} page The underlying model object from Sketch.
+   */
+  constructor(page = {}) {
+    if (!page.sketchObject) {
+      // eslint-disable-next-line no-param-reassign
+      page.sketchObject = Factory.createNative(Page)
+        .alloc()
+        .initWithFrame(new Rectangle(0, 0, 100, 100).asCGRect())
+    }
+
+    super(page)
+  }
+
+  /**
+   * The layers that the user has selected.
+   *
+   * @return {Selection} A selection object representing the layers that the user has selected.
+   */
+  get selectedLayers() {
+    return new Selection(this)
+  }
+
+  /**
+   * Export this layer (and the ones below it), using the options supplied.
+   *
+   * @param {dictionary} options Options indicating which layers to export, which sizes and formats to use, etc.
+   */
+  exportArtboards(options) {
+    const merged = { ...DEFAULT_EXPORT_OPTIONS, ...options }
+    const exporter = MSSelfContainedHighLevelExporter.alloc().initWithOptions(
+      merged
+    )
+    exporter.exportLayers(this.sketchObject.artboards())
+    return this
+  }
+
+  /* OVERRIDES */
+  export(options) {
+    const merged = { ...DEFAULT_EXPORT_OPTIONS, ...options }
+    const exporter = MSSelfContainedHighLevelExporter.alloc().initWithOptions(
+      merged
+    )
+    exporter.exportPage(this.sketchObject)
+    return this
+  }
+
+  // eslint-disable-next-line
+  adjustToFit() {
+    // obviously doesn't do anything
+  }
+
+  duplicate() {
+    const object = this._object
+    const duplicate = object.copy()
+    object.documentData().insertPage_afterPage(duplicate, object)
+    return wrapNativeObject(duplicate)
+  }
+
+  remove() {
+    this._object
+      .documentData()
+      .removePages_detachInstances([this._object], true)
+    return this
+  }
+
+  moveToFront() {
+    // doesn't do anything
```

Comment:
I know why this is here, but I wonder if it'd be cleaner to have Page extend a more lightweight class than Group, and use that also for Layer and Artboard… (I'm not sure about that either, just thinking aloud here…)

Reply:
yeah I think we will need to at some point. This can go in another PR tho, should be transparent